deltachat/rpc.go

```diff
@@ -45,20 +49,28 @@ type Rpc interface {
 
 // Delta Chat core RPC working over IO
 type RpcIO struct {
-	Stderr      io.Writer
-	AccountsDir string
-	Cmd         string
-	cmd         *exec.Cmd
-	stdin       io.WriteCloser
-	client      *jrpc2.Client
-	ctx         context.Context
-	events      map[AccountId]chan *Event
-	eventsMutex sync.Mutex
-	closed      bool
+	Stderr        io.Writer
+	AccountsDir   string
+	Cmd           string
+	EventBuffer   int
+	cmd           *exec.Cmd
+	stdin         io.WriteCloser
+	client        *jrpc2.Client
+	ctx           context.Context
+	cancel        context.CancelFunc
+	accountEvents map[AccountId]chan *Event
+	events        chan notifyParams
+	mu            sync.Mutex
 }
 
+var _ Rpc = &RpcIO{}
+
 func NewRpcIO() *RpcIO {
-	return &RpcIO{Cmd: "deltachat-rpc-server", Stderr: os.Stderr, closed: true}
+	return &RpcIO{
+		Cmd:         "deltachat-rpc-server",
+		Stderr:      os.Stderr,
+		EventBuffer: 10,
```

Comment:
This keeps the old behavior, somewhat, by queuing 10 events in/out per account. Isn't enough for tests but for a correctly implemented client or bot it should be enough.